### PR TITLE
Make classes hashable

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
Merge after https://github.com/orocos/orocos_kinematics_dynamics/pull/394

### CODE REVIEW

**Review by P-ict0 — DISMISSED** [review]

File: pykdl_ros/src/pykdl_ros/__init__.py [hunk 1]
```diff
@@ -41,6 +41,9 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
+    def __hash__(self):
+        return hash((self.frame, self.header.stamp, self.header.frame_id))
```
Comment (P-ict0) [inline pykdl_ros/src/pykdl_ros/__init__.py:45]: Shouldn't the __hash__() function only include the elements compared in __eq__()?

Comment (MatthijsBurgh) [reply pykdl_ros/src/pykdl_ros/__init__.py:45]: The python documentation says the following: `The only required property is that objects which compare equal have the same hash value`. So objects with a different timestamp will compare equal, but have a different hash. So you are right.

I don't think we want to include the timestamp in the compare function. So removing the stamp from the hash function is the best way to go IMO. Do you agree @P-ict0?

Comment (P-ict0) [reply pykdl_ros/src/pykdl_ros/__init__.py:45]: I think its the best option, yes

**Review by P-ict0 — APPROVED** [review]

**Comment by MatthijsBurgh** [comment]
@P-ict0 you should have waited for https://github.com/orocos/orocos_kinematics_dynamics/pull/394 to eb merged. That was in the main comment....
When you try to hash these classes now, you will still get an exception.

**Comment by P-ict0** [comment]
> @P-ict0 you should have waited for [orocos/orocos_kinematics_dynamics#394](https://github.com/orocos/orocos_kinematics_dynamics/pull/394) to eb merged. That was in the main comment.... When you try to hash these classes now, you will still get an exception.

Sorry, didn't notice. Will read better next time

**Pull request merged**